src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -3407,317 +3557,470 @@ private static string GetTimeZoneDirectory()
 
-            if (DstStartIndex >= DstEndIndex) {
-                // we found a DST start but no DST end
-                endDate = DateTime.MaxValue.Date;
+                    if (index < posixFormat.Length && posixFormat[index] == ',')
+                    {
+                        index++;
+                        TZif_ParsePosixDateTime(posixFormat, ref index, out end, out endTime);
+                    }
+                }
             }
 
-            AdjustmentRule prevRule = rulesList[ rulesList.Count - 1]; // grab the last element of the MultiYearRule sequence
-            int y = prevRule.DateEnd.Year;
-            if (prevRule.DateEnd.Month <= 6) {
-                // create a rule from 06/16/YYYY to endDate
-                AdjustmentRule r = AdjustmentRule.CreateAdjustmentRule(new DateTime(y, 06, 16), endDate, daylightBias, c_transition5_15, dstEnd);
-                rulesList.Add(r);
-            }
-            else {
-                // create a rule from 11/16/YYYY to endDate
-                AdjustmentRule r = AdjustmentRule.CreateAdjustmentRule(new DateTime(y, 11, 16), endDate, daylightBias, c_transition10_15, dstEnd);
-                rulesList.Add(r);
-            }
-        }    
+            return !string.IsNullOrEmpty(standardName) && !string.IsNullOrEmpty(standardOffset);
+        }
 
-        
-        static private void TZif_CreateMiddleMultiYearRules(ref List<AdjustmentRule> rulesList, TimeSpan daylightBias, DateTime endTransitionDate) {
-                    // 
-                    // [AdjustmentRule       #1] // middle-year all-DST rule
-                    // [1947/06/16 - 1947/11/15] // * starts 1 day after last day in previous rule
-                    // [start: 05/15           ] // * 5 months long, stopping at month 6 or 11
-                    // [end  : 12/15           ] // notice how the _start and end_ are outside the range
-                    //
-                    // [AdjustmentRule       #2] // middle-year all-DST rule
-                    // [1947/11/16 - 1947/06/15] //  * starts 1 day after last day in previous rule
-                    // [start: 10/01           ] //  * 7 months long, stopping at month 6 or 11
-                    // [end  : 07/15           ] // notice how the _start and end_ are outside the range
-                    //  
-                    // .........................
+        private static string TZif_ParsePosixName(string posixFormat, ref int index)
+        {
+            return TZif_ParsePosixString(posixFormat, ref index, c => char.IsDigit(c) || c == '+' || c == '-' || c == ',');
+        }
 
-            AdjustmentRule prevRule = rulesList[ rulesList.Count - 1]; // grab the first element of the MultiYearRule sequence
-            DateTime endDate;
+        private static string TZif_ParsePosixOffset(string posixFormat, ref int index)
+        {
+            return TZif_ParsePosixString(posixFormat, ref index, c => !char.IsDigit(c) && c != '+' && c != '-' && c != ':');
```

Comment:
This shouldn't matter in this situation for 2 reasons:
1. The string characters in the tzfile are all ASCII.
2. We will attempt to parse the string into a TimeSpan, and if that fails either an exception will occur or the rule will be skipped - depending on which code path is trying to parse the offset.